### PR TITLE
ManagementGroups SDK

### PR DESCRIPTION
**Pull request by rajshah11**
## Description

This PR adds the ManagementGroups SDK

Rest API specs PR - [https://github.com/Azure/azure-rest-api-specs/pull/2469](https://github.com/Azure/azure-rest-api-specs/pull/2469)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/ManagementGroups/Management.ManagementGroups/Management.ManagementGroups.csproj [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <!-- Please do not move/edit code below this line -->
+	<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+	<!-- Please do not move/edit code below this line -->
```
Comment (shahabhijeet) [inline src/SDKs/ManagementGroups/Management.ManagementGroups/Management.ManagementGroups.csproj]: change the word "below" to above

File: src/SDKs/ManagementGroups/ManagementGroups.Tests/InMemoryTests/ManagementGroups.InMemoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            HttpMockServer.Mode = HttpRecorderMode.Playback;
+            return client;
+        }
+
+        [Fact]
+        public void ListGroups()
+        {
+            var response = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = new StringContent(@"{
```
Comment (shahabhijeet) [inline src/SDKs/ManagementGroups/ManagementGroups.Tests/InMemoryTests/ManagementGroups.InMemoryTests.cs]: @rajshah11 why these responses hardcoded? why are they not getting recorded?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/ManagementGroups/ManagementGroups.Tests/InMemoryTests/ManagementGroups.InMemoryTests.cs (hunk 2 above)
Comment (rajshah11) [reply src/SDKs/ManagementGroups/ManagementGroups.Tests/InMemoryTests/ManagementGroups.InMemoryTests.cs]: These are InMemoryTests. All the scenarios are recorded here - [Scenario Tests](https://github.com/rajshah11/azure-sdk-for-net/blob/75490805e4f9742c6ac9ab60dc4e7118bbfb291c/src/SDKs/ManagementGroups/ManagementGroups.Tests/ScenarioTests/ManagementGroups.ScenarioTests.cs)

File: src/SDKs/ManagementGroups/Management.ManagementGroups/Management.ManagementGroups.csproj (hunk 1 above)
Comment (rajshah11) [reply src/SDKs/ManagementGroups/Management.ManagementGroups/Management.ManagementGroups.csproj]: Done. Thanks!

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**